### ISSUE TITLE
Add per account proxy settings (SOCKS, Tor, etc.)

### ISSUE THREAD

**Issue by dllud** [siacs/Conversations#1930]
Currently, as of version 1.13.2, Conversions only supports Tor on a global level, i.e., either all or none of the traffic gets routed through Tor. It would be much handy if this could be set on a per account basis. Desktop clients like Pidgin and Gajim, and even ChatSecure for Android, provide this feature.

With a per account approach, users could add accounts with different anonymity requirements, without harming usability, and with no increase on mobile data budgets. For instance, you could set up your top secret account to use Tor, while your public account connects via clearnet. This way, your public account would still fetch messages on networks that are either too slow, short lived or expensive (mobile data) for Tor.
Right now, the only sane solution is to have all accounts use Tor. This adds unnecessary network overhead to both your connection and the Tor network.

**Comment by d9h02f** [comment]
If you only send your top secret stuff through Tor, you make it very easy for anyone watching you to figure out what you're doing, thus defeating the point of using Tor.

**Comment by dllud** [comment]
It all depends on your threat model, i.e., what you are using Tor for.

If you adversary is your ISP or someone else who is able to record to all your network traffic then it makes sense to route everything through Tor.
On the other hand, if your threat model is more relaxed, for instance, you are just using Tor to avoid the recipient (e.g. a XMPP server) from knowing your real IP, then using it alongside a clearnet account won't be detrimental.

**Comment by GrimKriegor** [comment]
+1

**Comment by matthaab** [comment]
+1

**Comment by mglinux** [comment]
+1

**Comment by xpressengine3** [comment]
+1

**Comment by blevant** [comment]
socks5 greatly preferred 


else use iptables front end to constrain conversations to vpn connection

**Comment by griffindoors** [comment]
+10
SOCKS5

**Comment by gitsla** [comment]
+1
There are plenty of reasons why it would be very helpful to be able to connect to your accounts with different proxy settings. Please add this feature :)
Is it difficult to implement this? In my imagination it shouldn't be too hard or dangerous, but unfortunately I had no time yet to look into the code :/

**Comment by Robbilie** [comment]
+1

**Comment by licaon-kter** [comment]
FYI, using an .onion server would get that account (at least?) used over Tor.

**Comment by xeverse** [comment]
Essential..

**Comment by l-n-s** [comment]
Waiting for this to be added.. I have to use Telegram instead of XMPP, because it supports custom proxies.

**Comment by licaon-kter** [comment]
@iNPUTmice @voey

Heads-up, as of Orbot-16.1.0-BETA-4 the HTTP proxy port is no longer 8**1**18, but set up randomly on each Orbot start, while Conversations is hardcoded to try and connect to port 8**1**18.

Same for SOCKS and others, but at the moment you can setup the SOCKS port in Orbot back to 9050, but there is no such control for HTTP.

Looking at `netstat -tupan|grep LISTEN` I can see a port 8**2**18 (and 9**1**50 and 9**1**40) listening too and using the proxy settings with 127.0.0.1 and 8**2**18 connections work again in F-Droid (which has an actual proxy settings option).

**Comment by licaon-kter** [comment]
^^^ @n8fr8 How is an app supposed to cope with this randomness?

**Comment by n8fr8** [comment]
Two ways:

1) Orbot sends out broadcasts on status with the ports. You can see an example here of how to handle them, or just use our OrbotHelper class in the NetCipher library.
https://github.com/guardianproject/NetCipher/blob/master/libnetcipher/src/info/guardianproject/netcipher/proxy/OrbotHelper.java
https://github.com/guardianproject/NetCipher

2) A user can override in Orbot settings, to set the ports to a static value

**Comment by licaon-kter** [comment]
@n8fr8 AS of BETA-4 I don't see a way to override HTTP port specifically, did I miss it?

**Comment by n8fr8** [comment]
You may be right! Will make sure that gets in. We might also add a setting on the main screen to make it easy for people to set to static ports.

In the future, we either want apps to use the OrbotHelper/Broadcasts or for people to use the App VPN feature, thus static ports must end! :)

**Comment by n8fr8** [comment]
https://gitlab.com/guardianproject/orbot/issues/1

**Comment by n8fr8** [comment]
(using our new clean tracker on gitlab!)

**Comment by licaon-kter** [comment]
@n8fr8 the issue is gone?

Anyway, appears fixed in BETA5

**Comment by dllud** [comment]
IMHO you should reconsider the usage of this stale bot to close issues.
An issue should only be closed when it gets solved or it is no longer relevant. This issue, which is actually a feature request, reached none of those stages.
If you wish the bot to work as a reminder please disable it from closing issues. As it is, it just forces issue reporters to make pointless comments (as this one).
I had this same discussion on a few other projects. The general agreement was that this bot creates nothing but superfluous discussions.

**Comment by licaon-kter** [comment]
@dllud your comment didn't add anything of value

**Comment by dllud** [comment]
@licaon-kter Exactly. But I was forced to add a comment due to the stale bot. That's the whole point of my comment: disable this bot, as otherwise you'll get pointless comments just to keep an issue alive.

**Comment by iNPUTmice** [comment]
@dllud no you were not forced to write that comment. Those issues are here to help the project. If I don’t want stale issues around that I haven’t worked on in 4 years that's my decision.

**Comment by XutaxKamay** [comment]
This would be useful not only for Tor, but especially I2P with a tunnel socks5 proxy in order to connect to the XMPP server.

I think if nobody did it since years I'll give it a try myself.

**Comment by r4sas** [comment]
Conversations with I2P support: https://github.com/r4sas/Conversations-I2P

@iNPUTmice maybe you will add support?
